### DISCUSSION TITLE
Does silero-vad contain any noise filtering algorithm?

### DISCUSSION THREAD

**Discussion by balazsgonczy** [snakers4/silero-vad#172] [Ideas]
Hi,

I would like to apply your algo on a Speaker Identification task on noisy recordings. Have you tested it on noisy data? If not then what noise filtering algo do you recommend? I have already tried "nr.reduce_noise" but it slows down the process insanely.

Bw,

Balázs

**Comment by snakers4** [comment]
Hi,

We have designed VAD to work with reasonably noisy data. Typically it requires fiddling with hyper-params a bit like setting longer minimal speech duration and higher threshold.
As for noise filtering / audio enhancement - this is still work in progress for us.

**Reply by snakers4** [reply]
As usual, try plotting the chart with your data to see how well VAD performs. The VAD utils have a flag for that

**Comment by balazsgonczy** [comment]
Maybe can you recommend a library or a specific function? Sry I am totally
new to audio files, I used to working on tabular data and images. :D

Alexander Veysov ***@***.***> ezt írta (időpont: 2022. febr.
25., P 8:53):

**Reply by snakers4** [reply]
the only thing that comes to mind is a denoiser by facebook
but it has some limitations, like poor scripts in the repo, limited sampling rate, etc

**Comment by balazsgonczy** [comment]
Thx I think Ill only stick to ur algo, cuz its doing its job 99.75% of the
time.

Alexander Veysov ***@***.***> ezt írta (időpont: 2022. febr.
25., P 9:42):
